wallet-new/test/APISpec.hs

```diff
@@ -157,18 +162,39 @@ testV1Context =
     testRealModeContext :: IO (RealModeContext WalletMempoolExt)
     testRealModeContext = return (error "testRealModeContext is currently unimplemented")
 
+serverLayout :: ByteString
+serverLayout = Text.encodeUtf8 (layout (Proxy @V1.API))
+
 -- Our API apparently is returning JSON Arrays which is considered bad practice as very old
 -- browsers can be hacked: https://haacked.com/archive/2009/06/25/json-hijacking.aspx/
 -- The general consensus, after discussing this with the team, is that we can be moderately safe.
 spec :: Spec
 spec = withCompileInfo def $ do
     withDefConfigurations $ \_ -> do
-      xdescribe "Servant API Properties" $ do
-        it "V0 API follows best practices & is RESTful abiding" $ do
-          ddl <- D.dummyDiffusionLayer
-          withServantServer (Proxy @V0.API) (v0Server (D.diffusion ddl)) $ \burl ->
-            serverSatisfies (Proxy @V0.API) burl stdArgs predicates
-        it "V1 API follows best practices & is RESTful abiding" $ do
-          ddl <- D.dummyDiffusionLayer
-          withServantServer (Proxy @V1.API) (v1Server (D.diffusion ddl)) $ \burl ->
-            serverSatisfies (Proxy @V1.API) burl stdArgs predicates
+        xdescribe "Servant API Properties" $ do
+            it "V0 API follows best practices & is RESTful abiding" $ do
+                ddl <- D.dummyDiffusionLayer
+                withServantServer (Proxy @V0.API) (v0Server (D.diffusion ddl)) $ \burl ->
+                    serverSatisfies (Proxy @V0.API) burl stdArgs predicates
+            it "V1 API follows best practices & is RESTful abiding" $ do
+                ddl <- D.dummyDiffusionLayer
+                withServantServer (Proxy @V1.API) (v1Server (D.diffusion ddl)) $ \burl ->
+                    serverSatisfies (Proxy @V1.API) burl stdArgs predicates
+
+    describe "Servant Layout" $ do
+        let strip = reverse . dropWhile isSpace . reverse . dropWhile isSpace
+        root <- fmap strip . runIO $ readProcess "git" ["rev-parse", "--show-toplevel"] []
```

Comment:
this is almost certainly *not* the right way to do this; what is?

Reply:
@parsonsmatt Have a look at `HasCompileInfo` and more specifically to `Pos.Util.CompileInfo`. It uses `Given` (from the `reflection` package) to carry around the git revision 😉

Reply:
I need the repository directory. 

Looks like `git` isn't on CI, so it's failing there :\

Reply:
> I need the repository directory.

Ah! Do you think it's worth adding that to `HasCompileInfo` then? It might turn out to be useful not just here, and you would avoid the `runIO` thingy.

> Looks like git isn't on CI

Ops! Maybe is provisioned differently and is not in `$PATH`?